src/gwt/acesupport/acemode/r_code_model.js

```diff
@@ -761,30 +761,52 @@ var RCodeModel = function(session, tokenizer,
 
    this.$buildScopeTreeUpToRow = function(maxRow)
    {
-      function getChunkLabel(reOptions, comment) {
+      function getChunkLabel(reOptions, comment, iterator) {
 
          if (typeof reOptions === "undefined")
             return "";
          
          var match = reOptions.exec(comment);
-         if (!match)
-            return null;
-         var value = match[1];
-         var values = value.split(',');
-         if (values.length == 0)
-            return null;
-
-         // If first arg has no =, it's a label
-         if (!/=/.test(values[0])) {
-            return values[0].replace(/(^\s+)|(\s+$)/g, '');
+         if (match)
+         {
+            var value = match[1];
+            var values = value.split(',');
+            if (values.length > 0)
+            {
+               // If first arg has no =, it's a label
+               var first = values[0];
+               if (!/=/.test(first)) {
+                  var label = first.replace(/(^\s+)|(\s+$)/g, '');
+                  if (label.length)
+                     return label;
+               }
+            
+               for (var i = 0; i < values.length; i++) {
+                  match = /^\s*label\s*=\s*(.*)$/.exec(values[i]);
+                  if (match)
+                     return Utils.stripEnclosingQuotes(match[1].trim());
+               }
+            }
          }
-
-         for (var i = 0; i < values.length; i++) {
-            match = /^\s*label\s*=\s*(.*)$/.exec(values[i]);
-            if (match)
-               return Utils.stripEnclosingQuotes(match[1].trim());
+         
+         // then look for yaml label, i.e.
+         // 
+         // ```{r}
+         // #| label: foo
+         var it = iterator.clone();
+         
+         var tok = it.moveToNextToken();
+         while(tok != null && tok.type.startsWith("comment"))
+         {
+            var value = tok.value;
+            
+            var labelRegex = /^#\| *label *: *(.*)$/;
```

Comment:
I feel like we usually use `\s*` here rather than ` *` but I don't see why it would make a difference